test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -211,6 +213,116 @@ public void CreateDisplayMetadata_DisplayAttribute_DescriptionFromResources()
             Assert.Equal("description from resources", context.DisplayMetadata.Description());
         }
 
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_NullLocalizer()
+        {
+            // Arrange
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory: null);
+
+            var display = new DisplayAttribute()
+            {
+                Name = "Model_Name",
+                Description = "Model_Description",
+                Prompt = "Model_Prompt"
+            };
+
+            var attributes = new Attribute[] { display };
+            var key = ModelMetadataIdentity.ForType(typeof(DataAnnotationsMetadataProviderTest));
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+
+            // Act
+            provider.CreateDisplayMetadata(context);
+
+            // Assert
+            Assert.Equal("Model_Name", context.DisplayMetadata.DisplayName());
+            Assert.Equal("Model_Description", context.DisplayMetadata.Description());
+            Assert.Equal("Model_Prompt", context.DisplayMetadata.Placeholder());
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_LocalizerMissingProperties()
```

Comment:
I intend this test as insurance against anyone in the future trying to do something other than returning the key as value in the event that that's what the localizer returns.